public/app/features/auth-config/fields.tsx

```diff
@@ -620,6 +627,69 @@ export function fieldMap(provider: string): Record<string, FieldData> {
         'If enabled, Grafana will match the Hosted Domain retrieved from the Google ID Token against the Allowed Domains list specified by the user.',
       type: 'checkbox',
     },
+    serverDiscoveryUrl: {
+      label: 'OpenID Connect Discovery URL',
+      description:
+        'The .well-known/openid-configuration endpoint for your IdP. The info extracted from this URL will be used to populate the Auth URL, Token URL and API URL fields.',
+      type: 'custom',
+      content: (setValue, watch) => {
+        const modalIsOpen = watch('serverDiscoveryModal');
+        const onClose = () => setValue('serverDiscoveryModal', false);
+        const onSuccess = async (data: ServerDiscoveryFormData) => {
+          try {
+            const wellKnownSuffix = '/.well-known/openid-configuration';
+            const url = new URL(data.url);
+            if (!url.pathname.includes(wellKnownSuffix)) {
+              data.url = url.origin + wellKnownSuffix;
+            }
+
+            const response = await fetch(data.url);
+            const res = await response.json();
```

Comment:
You'd be able to use `getBackendSrv().get` here to not to have to parse the JSON manually.

Reply:
Done.